lib/jt400.ts

```diff
@@ -384,19 +365,65 @@ export interface InMemoryConnection extends Connection {
 	mockPgm: (programName: string, fn: (input: any) => any) => InMemoryConnection
 }
 
-export function pool(config?): Connection {
+export interface nodeJava {
+	classpath: Array<string>,
+	options: Array<string>,
+	nativeBindingLocation: string,
+	asyncOptions: object,
+	isJvmCreated: Function,
+	registerClient: Function,
+	registerClientP: Function,
+	ensureJvm: Function,
+	onJvmCreated: Function,
+	import: Function,
+};
+
+export function initialize(java?: nodeJava) {
+	const jvm = java ? java : require('java');
+	
+	const asyncOptions = {
+		asyncSuffix: "",
+		syncSuffix: "Sync",
+		promiseSuffix: "Promise", // Generate methods returning promises, using the suffix Promise.
+		promisify: promisify
+	};
+	// Do not set asyncOptions if it has already been set
+	// Allows for multiple instances to be started with passed java instance
+	if (!jvm.asyncOptions) {
+		jvm.asyncOptions = asyncOptions;
+	}
+	
+	jvm.options.push('-Xrs'); // fixing the signal handling issues (for exmaple ctrl-c)
+
+	jvm.classpath.push(__dirname + '/../../java/lib/jt400.jar');
+	jvm.classpath.push(__dirname + '/../../java/lib/jt400wrap.jar');
+	jvm.classpath.push(__dirname + '/../../java/lib/json-simple-1.1.1.jar');
+	jvm.classpath.push(__dirname + '/../../java/lib/hsqldb.jar');
+
+	process.on('exit', function(code) {
+		jvm.import('java.lang.System').exit(code);
+	});
+
+	return jvm;
+}
+
+export function pool(config?, java?: nodeJava): Connection {
+	const jvm = initialize(java);
```

Comment:
I thought about doing that but I only call pool once and export it. So I thought it would be an unneeded extra step.

Here is a snippet of what I do typically in my projects.

#### javaInit.js
```javascript
'use strict';

const java = require('java');
const jt400 = require('node-jt400');
const config = require('./config');

// Any custom java programs get put under java directory
// Any and all have to be initilized here
java.classpath.push(`${__dirname}/java`);
const iseries = jt400.pool(config, java);

// Custom node java applications
const customApp = java.import('com.testing.custom.app');

module.exports = {
        customApp,
	iseries,
};
```
I then run the functions on the exported pool.
```javascript
'use strict';
const { iseries } = require('./javaInit');

const getData = () => {
    try {
        const sql = 'select current date from sysibm.sysdummy1';
        const data = await iseries.query(sql);
        return data;
    }
    catch (err) {
        throw err;
    }
};
```

Reply:
Yes, but that is just one use case. It is possible to call pool multiple times, for example if you need connection to more than one system, and then java will be initialised each time.

Reply:
Makes sense. See latest commit.